src/qt/bitcoingui.cpp

```diff
@@ -1452,23 +1452,72 @@ void BitcoinGUI::updateScraperIcon(int scraperEventtype, int status)
 
     int64_t nConvergenceTime = ConvergedScraperStatsCache.nTime;
 
-    std::string sExcludedProjects = {};
+    std::string sExcludedProjects;
+    std::string sIncludedScrapers;
+    std::string sExcludedScrapers;
+    std::string sScrapersNotPublishing;
 
-    bool bExcludedProjects = false;
+    bool bDisplayScrapers = false;
 
     // If the convergence cache has excluded projects...
-    if (ConvergedScraperStatsCache.vExcludedProjects.size())
+    if (!ConvergedScraperStatsCache.Convergence.vExcludedProjects.empty())
     {
-        bExcludedProjects = true;
-
-        for (const auto& iter : ConvergedScraperStatsCache.vExcludedProjects)
+        for (const auto& iter : ConvergedScraperStatsCache.Convergence.vExcludedProjects)
         {
             if (sExcludedProjects.empty())
                 sExcludedProjects += iter.first;
             else
                 sExcludedProjects += ", " + iter.first;
         }
     }
+    else
+    {
+        sExcludedProjects = "none";
+    }
+
+    // If fDebug3 then show scrapers in tooltip...
+    if (fDebug3)
+    {
+        bDisplayScrapers = true;
+
+        for (const auto& iter : ConvergedScraperStatsCache.Convergence.vIncludedScrapers)
```

Comment:
If you want to you can replace these 3 loops with calls to [boost join](https://www.boost.org/doc/libs/1_41_0/doc/html/string_algo/reference.html#header.boost.algorithm.string.join_hpp). Check [here](https://stackoverflow.com/a/1833499) for an example.

Reply:
Ooh... I didn't know about that one. I will! :)